File: smart_farm/src/api/deviceType.ts
import { get, http, post } from '@/utils/request'
import type { PageInfo } from '@/api/user'

/**
 * 设备类型实体，对应 smart_plant 后端 device_type 表字段。
 * 页面统一使用该类型承接列表、详情和状态切换后的返回数据。
 */
export interface DeviceType {
  id?: number
  typeCode: string
  typeName: string
  category: number
  description?: string
  status?: number
  createTime?: string
  updateTime?: string
}

/**
 * 新增设备类型时提交给后端的字段。
 * id、createTime、updateTime 由数据库和后端维护，不在新增表单中提交。
 */
export interface DeviceTypePayload {
  typeCode: string
  typeName: string
  category: number
  description?: string
  status?: number
}

export interface UpdateDeviceTypePayload extends Partial<DeviceTypePayload> {
  id: number
}

export interface DeviceTypeListParams {
  typeCode?: string
  typeName?: string
  category?: number
  status?: number
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

/** 分页查询设备类型列表，对接 smart_plant DeviceTypeController -> GET /device-type/list。 */
export const listDeviceTypes = (params: DeviceTypeListParams) => {
  return get<ApiResult<PageInfo<DeviceType>>>(
    '/device-type/list',
    params as unknown as Record<string, unknown>,
  )
}

/** 根据设备类型 ID 查询详情，供编辑弹框回填表单使用。 */
export const getDeviceTypeById = (id: number) => {
  return get<ApiResult<DeviceType>>(`/device-type/${id}`)
}

/** 新增设备类型，对接 POST /device-type/add。 */
export const addDeviceType = (data: DeviceTypePayload) => {
  return post<ApiResult<DeviceType>>('/device-type/add', data as unknown as Record<string, unknown>)
}

/** 编辑设备类型基础信息，对接 PUT /device-type。 */
export const updateDeviceType = (data: UpdateDeviceTypePayload) => {
  return http<ApiResult<DeviceType>>({ url: '/device-type', method: 'put', data })
}

/** 删除单个设备类型，后端会阻止删除已被设备引用的类型。 */
export const deleteDeviceType = (id: number) => {
  return http<ApiResult<void>>({ url: `/device-type/${id}`, method: 'delete' })
}

/** 批量删除设备类型。 */
export const batchDeleteDeviceTypes = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/device-type/batch', method: 'delete', data: ids })
}

/** 启用或禁用设备类型，status 为 1 表示启用，0 表示禁用。 */
export const updateDeviceTypeStatus = (id: number, status: number) => {
  return http<ApiResult<void>>({
    url: `/device-type/${id}/status`,
    method: 'put',
    params: { status },
  })
}

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/DeviceType.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

/**
 * 设备类型实体类。
 *
 * <p>该类与数据库 device_type 表对应，用于维护传感器、执行器等设备类型基础信息。</p>
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DeviceType {

    /** 设备类型主键ID，由数据库自增生成。 */
    private Long id;

    /** 类型编码，全表唯一，用于前端选项、设备接入或第三方系统识别。 */
    private String typeCode;

    /** 类型名称，用于页面展示，例如环境传感器、水泵。 */
    private String typeName;

    /** 设备分类，1表示传感器，2表示执行器。 */
    private Integer category;

    /** 类型说明，描述该类设备的用途或采集/控制能力。 */
    private String description;

    /** 状态，1启用，0禁用；禁用后不允许新建设备继续选择该类型。 */
    private Integer status;

    /** 记录创建时间，由数据库默认写入。 */
    private LocalDateTime createTime;

    /** 记录更新时间，由数据库自动维护。 */
    private LocalDateTime updateTime;
}
